tests/radv/test_radv_ipv6_ra.py

```diff
@@ -0,0 +1,158 @@
+        downlink_vlan_iface['ip6'] = str(ip6)
+
+        # Obtain link-local IPv6 address of the connected PTF port (Eg eth0)
+        # This PTF port maps to the first member of the TOR's VLAN
+        ptf_port = {}
+        ptf_port['port_idx'] = mg_facts['minigraph_ptf_indices'][vlan_info_dict['members'][0]]
+        ptf_port['name'] = "eth" + str(ptf_port['port_idx'])
+        cmd = "ip -6 -o addr show dev {} scope link | sed -e's/^.*inet6 \([^ ]*\)\/.*$/\\1/;t;d'".format(ptf_port['name'])
+        res = ptfhost.shell(cmd)
+        ip6 = ipaddress.IPv6Address(unicode(res['stdout']))
+        pytest_assert(ip6.is_link_local,
+                "ip6 address:{} of {} is not a link-local address".format(str(ip6), ptf_port['name']))
+        ptf_port['ip6'] = str(ip6)
+
+        vlan_intf_data = {}
+        vlan_intf_data['downlink_vlan_intf'] = downlink_vlan_iface
+        vlan_intf_data['ptf_port'] = ptf_port
+        vlan_interfaces_list.append(vlan_intf_data)
+
+    return vlan_interfaces_list
+
+"""
+@summary: Updates min/max RA interval in RADVd's config file
+
+"""
+
+def dut_update_ra_interval(duthost, ra, interval):
+    logging.info("Updating %s to %d in RADVd's config file:%s", ra, int(interval), RADV_CONF_FILE)
+    cmd = "sed -ie 's/\(.*\)\({}\) \([[:digit:]]\+\)/\\1\\2 {}/' {}".format(ra, interval, RADV_CONF_FILE)
+    duthost.shell("docker exec radv {}".format(cmd))
+
+"""
+@summary: A fixture that updates the RADVd's periodic RA update intervals and restores the
+intervals to old values after the test
+
+"""
+@pytest.fixture
+def dut_update_radv_periodic_ra_interval(duthost):
+    pytest_assert(duthost.is_service_fully_started('radv'), "radv service not running")
+
+    cmd = 'docker exec radv [ -f {} ] && echo "1" || echo "0"'.format(RADV_CONF_FILE)
+    pytest_assert(u'1' == duthost.shell(cmd)["stdout"], "radv conf file {} NOT found".format(RADV_CONF_FILE))
+
+    cmd = "grep 'MinRtrAdvInterval' {}".format(RADV_CONF_FILE)
+    cmd += " | sed -e 's/.*\(MinRtrAdvInterval\) \([[:digit:]]\+\)\;/\\2/;'"
+    curr_min_ra_interval = duthost.shell('docker exec radv {}'.format(cmd))["stdout"]
+
+    cmd = "grep 'MaxRtrAdvInterval' {}".format(RADV_CONF_FILE)
+    cmd += " | sed -e 's/.*\(MaxRtrAdvInterval\) \([[:digit:]]\+\)\;/\\2/;'"
+    curr_max_ra_interval = duthost.shell('docker exec radv {}'.format(cmd))["stdout"]
+
+    dut_update_ra_interval(duthost, "MinRtrAdvInterval", RADV_MIN_RA_INTERVAL)
+    dut_update_ra_interval(duthost, "MaxRtrAdvInterval", RADV_MAX_RA_INTERVAL)
+
+    # Notify RADVd to read the updated RA intervals
+    logging.info("Notifying RADVd to read the updated config file:%s", RADV_CONF_FILE)
+    duthost.shell("docker exec radv supervisorctl signal SIGHUP radvd")
+
+    yield
+    # Restore the RA interval back to original values
+    dut_update_ra_interval(duthost, "MinRtrAdvInterval", curr_min_ra_interval)
+    dut_update_ra_interval(duthost, "MaxRtrAdvInterval", curr_max_ra_interval)
```

Comment:
Rather than storing the original values and setting them back, I suggest simply copying the file to create a backup before making the modifications above. Then here simply overwriting the modified file with the copy of the original.

Reply:
Ok . What is the advantage of copying over in-place modification of the file.

Reply:
Less operations, plus you can guarantee you are reverting the file back to its original state. Sometimes we've seen in-place sed calls corrupt files.